loki_operator.md

```diff
@@ -0,0 +1,123 @@
+
+Example with AWS S3, using `aws` CLI:
+
+```bash
+S3_NAME="netobserv-loki"
+AWS_REGION="eu-west-1"
+AWS_KEY=$(aws configure get aws_access_key_id)
+AWS_SECRET=$(aws configure get aws_secret_access_key)
+
+aws s3api create-bucket --bucket $S3_NAME  --region $AWS_REGION --create-bucket-configuration LocationConstraint=$AWS_REGION
+
+kubectl create -n netobserv secret generic lokistack-dev-s3 \
+  --from-literal=bucketnames="$S3_NAME" \
+  --from-literal=endpoint="https://s3.${AWS_REGION}.amazonaws.com" \
+  --from-literal=access_key_id="${AWS_KEY}" \
+  --from-literal=access_key_secret="${AWS_SECRET}" \
+  --from-literal=region="${AWS_REGION}"
+```
+
+## LokiStack
+
+Then create a `LokiStack` in `netobserv` namespace. When using OpenShift, navigate to:
+
+Administrator view -> Operators -> Installed Operators -> Loki Operator -> LokiStack -> Create LokiStack
+
+- Name it `loki` (any name is fine, but you need to adapt the URLs below accordingly)
+- Choose the size. While not suitable for production, `1x.extra-small` is OK for testing / demo. Note that very small clusters (e.g. 3 worker nodes) require `1x.extra-small`, see [troubleshooting](#troubleshooting) section below.
+- Set `Object Storage` -> `Secret` as noted above.
+- Set `Tenants Configuration` -> `Mode` to `openshift-network`. As a temporary hack, until `openshift-network` is available (expected in 5.6+), you can use `openshift-logging` instead. Just don't do that in production.
+
+This will create `gateway`, `distributor`, `compactor`, `ingester`, `querier` and `query-frontend` components.
+
+To allow `flowlogs-pipeline` to write to the gateway and `network-observability-plugin` to read from the gateway, you will need to create related `ClusterRole` and `ClusterRoleBinding` using:
+
+```bash
+kubectl apply -f examples/loki-stack/role.yaml
+# or as a temporary hack until `openshift-network` tenant mode is available (expected in 5.6+), use "examples/loki-stack/role-hack.yaml" instead.
+```
+
+## NetObserv configuration
+
+Once the Loki stack is up and running, you need to configure NetObserv to communicate to Loki through its `gateway` service. Loki CA certificate must have been written in a configmap, so it will be used for TLS.
+
+Then you will be able to set the following configuration in `FlowCollector` for `network` tenant:
+
+```yaml
+  loki:
+    url: 'https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/network/'
+    # temporary hack: for infrastructure tenant, use 'https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/infrastructure/' instead
+    statusUrl: 'https://loki-query-frontend-http.netobserv.svc:3100'
+    tenantID: network
+    # temporary hack: for infrastructure tenant, use infrastructure
+    authToken: HOST
+    tls:
+      enable: true
+      caCert:
+        type: configmap
+        name: loki-ca-bundle
```

Comment:
would below way to get CA bundle still be correct?
```yaml
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: loki-ca-bundle
  annotations:
    service.beta.openshift.io/inject-cabundle: "true"
```

Reply:
the configmap is automatically created by the loki operator, I didn't have to do anything to get it

Reply:
no... probably because I install LokiStack in netobserv NS so I don't have to copy them